lib/lightning/credentials.ex

```diff
@@ -221,16 +230,19 @@ defmodule Lightning.Credentials do
 
   defp cast_body_change(changeset), do: changeset
 
+  # TODO - FOR DISUSSION - does this mean that we can't use booleans or integers in raw credentials anymore?
   defp put_typed_body(body, "raw"), do: {:ok, body}
 
+  # TODO - roger i think we need tests for this function
   defp put_typed_body(body, schema_name) do
     schema = get_schema(schema_name)
 
     with changeset <- SchemaDocument.changeset(body, schema: schema),
          {:ok, typed_body} <- Ecto.Changeset.apply_action(changeset, :insert) do
       updated_body =
         Enum.into(typed_body, body, fn {field, typed_value} ->
-          {to_string(field), typed_value}
+          # TODO - roger i removed the "to string here" and it seems to work better
```

Comment:
we need to convert the keys resulting from `apply_action` from atom to string once the attributes comes from the form with string keys